app/controllers/repos_controller.rb

```diff
@@ -27,8 +27,9 @@ def show
       @charts = @benchmark_type.benchmark_result_types.map do |benchmark_result_type|
         cache_key = "#{BenchmarkRun.charts_cache_key(@benchmark_type, benchmark_result_type)}:#{@benchmark_run_display_count}"
 
-        if (columns = $redis.get(cache_key))
-          [JSON.parse(columns).symbolize_keys!, benchmark_result_type]
+        if (cache_read_json = $redis.get(cache_key))
+          cache_read = JSON.parse(cache_read_json, symbolize_names: true)
+          ChartBuilder.new([], benchmark_result_type).construct_from_cache(cache_read)
```

Comment:
you can make `construct_from_cache` a class method, then do all the work in there.
```ruby
def self.construct_from_cache(cache_read, type)
  c = ChartBuilder.new([], type)
  c.version = ...
  ...
  c
end
```